iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -534,34 +564,24 @@ public boolean canProvideColStatistics(org.apache.hadoop.hive.ql.metadata.Table
   }
 
   private boolean canProvideColStats(Table table, long snapshotId) {
-    Path statsPath = getColStatsPath(table, snapshotId);
-    try {
-      FileSystem fs = statsPath.getFileSystem(conf);
-      return  fs.exists(statsPath);
-    } catch (Exception e) {
-      LOG.warn("Exception when trying to find Iceberg column stats for table:{} , snapshot:{} , " +
-          "statsPath: {} , stack trace: {}", table.name(), table.currentSnapshot(), statsPath, e);
-    }
-    return false;
+    return IcebergTableUtil.getColStatsPath(table, snapshotId).isPresent();
```

Comment:
Do we know in one query how many times do we call this ``getColStatsPath()``? I think atleast twice, once for `canProvideColStats` & second would be when we do `getColStatistics`, if more or not that cheap then may be we should explore caching it or some other way to avoid calculating multiple times

Reply:
it's not doing FS listing, just iceberg stats metadata filter

Reply:
That doesn't read the Stats file? If not thats cool, it should be cheap if no FS call is involved

Reply:
no, just checks metadata if it has stats refs there